### ISSUE TITLE
CUDA packages are ~50MB

### ISSUE THREAD

**Issue by jaimergp** [conda-forge/openmm-feedstock#2]
`omnia` packages used to be <15MB, but our builds are sitting in the 50MB range. We are not even packaging docs, so investigate if we are packaging more things (copies?) than needed.

**Comment by jaimergp** [comment]
Differences come from the `*.so` files. `conda-forge` produces larger libraries than `omnia` builds:

```
     8237328  lib/libOpenMMAmoeba.so
      692232  lib/libOpenMMAmoeba.so
     1395264  lib/libOpenMMDrude.so
       89936  lib/libOpenMMDrude.so
      941440  lib/libOpenMMRPMD.so
       73504  lib/libOpenMMRPMD.so
    58769088  lib/libOpenMM.so
     4167333  lib/libOpenMM.so
     4893832  lib/plugins/libOpenMMAmoebaCUDA.so
     1163952  lib/plugins/libOpenMMAmoebaCUDA.so
     5945736  lib/plugins/libOpenMMAmoebaReference.so
      729224  lib/plugins/libOpenMMAmoebaReference.so
     9142360  lib/plugins/libOpenMMCPU.so
      655240  lib/plugins/libOpenMMCPU.so
     1078032  lib/plugins/libOpenMMCudaCompiler.so
       37096  lib/plugins/libOpenMMCudaCompiler.so
    21134576  lib/plugins/libOpenMMCUDA.so
     2742408  lib/plugins/libOpenMMCUDA.so
     1477200  lib/plugins/libOpenMMDrudeCUDA.so
      111912  lib/plugins/libOpenMMDrudeCUDA.so
     1662464  lib/plugins/libOpenMMDrudeOpenCL.so
      125000  lib/plugins/libOpenMMDrudeOpenCL.so
      812368  lib/plugins/libOpenMMDrudeReference.so
       72608  lib/plugins/libOpenMMDrudeReference.so
    22648072  lib/plugins/libOpenMMOpenCL.so
     2992112  lib/plugins/libOpenMMOpenCL.so
      695104  lib/plugins/libOpenMMPME.so
       80808  lib/plugins/libOpenMMPME.so
     1709432  lib/plugins/libOpenMMRPMDCUDA.so
      118744  lib/plugins/libOpenMMRPMDCUDA.so
     1960688  lib/plugins/libOpenMMRPMDOpenCL.so
      131856  lib/plugins/libOpenMMRPMDOpenCL.so
      706016  lib/plugins/libOpenMMRPMDReference.so
       67544  lib/plugins/libOpenMMRPMDReference.so
      632080  lib/python3.7/site-packages/simtk/openmm/app/internal/compiled.cpython-37m-x86_64-linux-gnu.so
      714624  lib/python3.7/site-packages/simtk/openmm/app/internal/compiled.cpython-37m-x86_64-linux-gnu.so
    14991896  lib/python3.7/site-packages/simtk/openmm/_openmm.cpython-37m-x86_64-linux-gnu.so
    14409920  lib/python3.7/site-packages/simtk/openmm/_openmm.cpython-37m-x86_64-linux-gnu.so
```

**Comment by jaimergp** [comment]
Some files can be 10x larger! Is there any compiler flags we are missing? IIRC, `omnia` uses `clang` but we are using `gcc`.

Any hints @peastman, @isuruf?

**Comment by jaimergp** [comment]
For reference, these are the CFLAGS and CPPFLAGS as reported in the build logs:

__Conda-forge__

The `gcc` sets the following env vars, which we augment with `-g -O3 -ldl` in `build.sh`.

```
CC=$BUILD_PREFIX/bin/x86_64-conda_cos6-linux-gnu-cc
CFLAGS=-march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -I$PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/openmm-7.4.0 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix

CPP=$BUILD_PREFIX/bin/x86_64-conda_cos6-linux-gnu-cpp
CPPFLAGS=-DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -I$PREFIX/include
CXXFLAGS=-fvisibility-inlines-hidden -std=c++17 -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -I$PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/openmm-7.4.0 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix
```

__Omnia__

I think [these only use `-g -O3`](https://github.com/omnia-md/conda-recipes-cuda/blob/master/openmm/build.sh#L18) (plus whatever the defaults are in the Docker image, which is based on the Conda-Forge Linux Anvil).

**Comment by jakirkham** [comment]
You're welcome to add more compilation flags in your build if that would help.

Just do `export CFLAGS="${CFLAGS} <some other stuff here>"` for example.

**Comment by jaimergp** [comment]
Closed by #6
